.travis.yml

```diff
@@ -26,6 +27,7 @@ script:
   - travis_retry goveralls -coverprofile=cover.out -service=travis-ci || true
 
 after_success:
+  - if [ "$TRAVIS_GO_VERSION" == "1.8" ]; then echo 'only build all-in-one docker image for go 1.7' ; exit 0; fi
```

Comment:
this is hacky. I would prefer if we used a matrix ([example](https://github.com/uber/tchannel-go/blob/dev/.travis.yml#L14)) and mark only one of the entries with an env variable that allows the after-success work. I also think that coveralls above should be in after-success.

also, how about moving all of the below to a shell script somewhere under `.travis` directory? Then we can have explicit env vars in the matrix, e.g. COVERAGE=1 ALL_IN_ONE=1 and instead of doing `exit` here we'd instead do

```
  if [ "$COVERAGE" == "1" ]; then ... ; else echo "skipping coverage"; fi
  if [ "$ALL_IN_ONE" == "1" ]; then ./travis/build-all-in-one-image.sh; else echo "skipping all-in-one"; fi
```